api/envoy/extensions/filters/http/credential_injector/v3/credential_injector.proto

```diff
@@ -0,0 +1,85 @@
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Credential Injector]
+// [#not-implemented-hide:]
+// Credential Injector :ref:`configuration overview <config_http_filters_credential_injector>`.
+// [#extension: envoy.filters.http.credential_injector]
+
+// Credential Injector injects credentials into outgoing HTTP requests. The filter configuration is used to retrieve the credentials, or
+// they can be requested through the OAuth2 client credential grant. The credentials obtained are then injected into the Authorization header
+// of the proxied HTTP requests, utilizing either the Basic or Bearer scheme.
+//
+// If the credential is not present, the request will fail with 401 Unauthorized if fail_if_not_present is set to true.
+//
+// Notice: This filter is intended to be used for workload authentication, which means that the identity associated with the inserted credential
+// is considered as the identity of the workload behind the envoy proxy(in this case, envoy is typically deployed as a sidecar alongside that
+// workload). Please note that this filter does not handle end user authentication. Its purpose is solely to authenticate the workload itself.
+//
+// Here is an example of CredentialInjector configuration with Generic credential, which injects an HTTP Basic Auth credential into the proxied requests.
+//
+//  .. code-block:: yaml
+//
+//  overwrite: true
+//  fail_if_not_present: true
+//  credential:
+//    name: generic_credential
+//    typed_config:
+//      "@type": type.googleapis.com/envoy.extensions.injected_credentials.generic.v3.Generic
+//      credential:
+//        name: credential
+//        sds_config:
+//          path_config_source:
+//            path: credential.yaml
+//      header: Authorization
+//
+// credential.yaml for Basic Auth:
+//  .. code-block:: yaml
+//
+//  resources:
+//  - "@type": "type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.Secret"
+//    name: credential
+//    generic_secret:
+//      secret:
+//        inline_string: "Basic base64EncodedUsernamePassword"
+//
+// It can also be configured to inject a Bearer token into the proxied requests.
+// credential.yaml for Bearer Token:
+//  .. code-block:: yaml
+//
+//  resources:
+//  - "@type": "type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.Secret"
+//    name: credential
+//    generic_secret:
+//      secret:
+//        inline_string: "Bearer myToken"
+//
+message CredentialInjector {
+  // Whether to overwrite the value or not if the injected headers already exist.
+  // Value defaults to false.
+  bool overwrite = 1;
+
+  // Whether to fail the request if the credential is not present.
+  // Value defaults to false.
+  // If set to true, the request will fail with 401 Unauthorized if the credential is not present.
+  bool fail_if_not_present = 2;
+
+  // The credential to inject into the proxied requests
+  // [#extension-category: envoy.injected_credentials]
```

Comment:
I think the CI error is because there are no injected_credentials extensions.
You can probably add a TODO comment tag here to add this later.